docs/spec/lightclient/verification/Blockchain_A_1.tla

```diff
@@ -75,7 +75,7 @@ LBT == [header |-> BT, Commits |-> {NT}]
 
 (* the header is still within the trusting period *)
 InTrustingPeriod(header) ==
-    now <= header.time + TRUSTING_PERIOD
+    now < header.time + TRUSTING_PERIOD
```

Comment:
I see the `<=` actually appears to contradict the [english spec](https://github.com/informalsystems/tendermint-rs/blob/master/docs/spec/lightclient/verification/verification.md#L424-L425) which says

>      - *LightStore* always contains a verified header whose age is less than the
>      trusting period,  

So it seems clear that what we ant is "less than", but perhaps this inconsistency was due in part to the wording and terminology here. E.g., in [this paragraph](https://github.com/informalsystems/tendermint-rs/blob/master/docs/spec/lightclient/verification/verification.md#L914-L916)

> In order to ensure liveness, *LightStore* always needs to contain a
> verified (or initially trusted) header whose time is within the
> trusting period.

"trust period" is spoken of as a duration that the header needs to be within *not* as a limit point that the header time needs to be *less than*.  I wonder if calling this `TRUST_PERIOD_LIMIT` might help prevent such confusion i the future. I suspect a problem with the current wording may arise due to "trusting period" sounds like a span within which things are valid, when it's actually meant to be the outer limit beyond which things are *invalid*. iiuc, the trust period is that span of time up to but not including the limit.

Reply:
You are right: `<=` does contradict the English spec, so this is the TLA+ modeling artifact. That there can be ambiguities in the English spec, that's of course possible -- English is not a formal language after all;) I still think that _our_ English spec does an amazing job of clarifying the requirements at the semi-formal (and as close to formal as possible) level. But exactly because of this non-strictness of the English language the TLA+ spec should, in my opinion, be the ultimate source of truth for all parties.

One of the things we were discussing with @josef-widder and @konnov is that now, when we have the three artifacts: the English spec, the TLA+ spec, and the implementation, and also now that we have the MBT, which serves as the "missing link" helping to synchronize them, we need to establish a _process_ (in the VDD sense) that will help us to resolve the conflicts. 

In this PR I've fixed the TLA+ spec, because it was the most easy way for me to go, but it's not clear at all whether that's the right thing to do. We need to figure out a joint understanding between the research and the development teams on how to proceed in such cases.

Reply:
In this case, given the larger context, I don't believe it matters much whether we use < or <= here since we are dealing with time and already assume clocks can drift some amount. The important thing is that `TRUST_PERIOD` is a fraction of the `UNBONDING_PERIOD`. This ensures that light clients have the duration of `UNBONDING_PERIOD - TRUST_PERIOD` to try and submit evidence to the chain and get faulty validators punished. And since all these periods are on the order of days/weeks, the < vs. <= here shouldn't make a difference in the end.

Also while UNBONDING_PERIOD is a global consensus parameter (all nodes agree on), the TRUST_PERIOD is not, it's a local subjective parameter that clients can set depending on their risk tolerance. So again, the plank second difference between `<` and `<=` shouldn't matter there.

Of course that might be an unsatisfactory answer - I don't mean to be flippant about it and we should certainly be consistent and precise, but it's important to keep the practical context in mind so we know what details are most important to focus on.

Reply:
Thanks for the feedback! I totally agree, from the practical perspective these differences between `<` and `<=` do not matter at all in this particular context, so this case we could fix however we like, and it would make no real difference. There are two things to keep in mind though:
- scientifically, we can't be approximately precise. And I would like to think about our specs as being precise, so we should clarify and fix any ambiguities.
- this particular example is only a manifestation of a general problem about the artifacts we are having, and I am sure we will encounter more instances of it, with probably higher level of severity. So it's better if we are prepared for that. 

Also I tend to perceive us having this problem as rather a good thing, demonstrating that from now on we will have a _living specification_, which is in sync with the implementation.

This is only the input from another feeble human mind, just a bit different;) And I am sure fixing this will not take much of either our focus or time.